File: Java/BridgePattern/src/resources/PodcastResource.java
package resources;

import models.Podcast;

/**
 * Author: Utkarsh Dayal @ descifrador17
 * Date: 13-06-2021
 * Time: 08:53 PM
 */
public class PodcastResource implements IResource{

    Podcast podcast;

    public PodcastResource(Podcast podcast) {
        this.podcast = podcast;
    }

    @Override
    public String snippet() {
        return podcast.getPodcastAbstract();
    }

    @Override
    public String title() {
        return podcast.getPodcastName();
    }

    @Override
    public String image() {
        return podcast.getPodcastImage();
    }

    @Override
    public String url() {
        return podcast.getPodcastLink();
    }
}

File: Kotlin/BridgePattern/src/resources/PodcastResource.kt
package resources

import models.Podcast

/**
 *
 *   Author: Utkarsh Dayal @ descifrador17
 *   Date: 13-06-2021
 *   Time: 09:34 PM
 *
 */
class PodcastResource(private val podcast: Podcast): IResource {
    override fun snippet(): String {
        return podcast.podcastAbstract
    }

    override fun title(): String {
        return podcast.podcastName
    }

    override fun image(): String {
        return podcast.podcastImage
    }

    override fun url(): String {
        return podcast.podcastLink
    }
}